File: extract_api.js
const fs = require('fs');
const path = require('path');

const res = new Set();
const pattern = /api\.(get|post|put|patch|delete)\(([`'"].*?[`'"])/g;
const srcDir = 'p:\\HOMESQRE CLONE\\homesqre\\frontend\\src';

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (/\.(js|jsx|ts|tsx)$/.test(file)) {
            const content = fs.readFileSync(fullPath, 'utf8');
            let match;
            while ((match = pattern.exec(content)) !== null) {
                res.add(`${match[1].toUpperCase()} ${match[2]}`);
            }
        }
    }
}

walkDir(srcDir);
const sorted = Array.from(res).sort();
fs.writeFileSync('p:\\HOMESQRE CLONE\\homesqre\\frontend_api_calls.txt', sorted.join('\n'));

File: extract_api.py
import os, re

res = set()
pattern = re.compile(r'api\.(get|post|put|patch|delete)\(([`\'\"].*?[`\'\"])')
src_dir = r'p:\HOMESQRE CLONE\homesqre\frontend\src'

for root, _, files in os.walk(src_dir):
    for file in files:
        if file.endswith(('.js', '.jsx', '.ts', '.tsx')):
            with open(os.path.join(root, file), 'r', encoding='utf-8') as f:
                content = f.read()
                matches = pattern.findall(content)
                for method, url in matches:
                    res.add(f'{method.upper()} {url}')

with open(r'p:\HOMESQRE CLONE\homesqre\frontend_api_calls.txt', 'w') as out:
    for i in sorted(list(res)):
        out.write(f"{i}\n")
